pydantic/env_settings.py

```diff
@@ -44,19 +45,58 @@ def _build_values(
         _env_file_encoding: Optional[str] = None,
         _secrets_dir: Union[Path, str, None] = None,
     ) -> Dict[str, Any]:
+        extra_settings = [source(self) for source in reversed(self.__config__.extra_settings_sources)]
```

Comment:
why the `reverse`?

Also shouldn't this be
```suggestion
        extra_settings = [self.load_env_vars_from_source(source) for source in reversed(self.__config__.extra_settings_sources)]
```
?

I think this would reduce the logic and complexity in your customer loaders/sources.

Reply:
As explained in my latest comment, the settings source sequence is expressed from the highest priority to the lowest one.
We need to reverse it to send the settings dicts to `deep_update` from the lowest priority to the highest one.

In the next revision, I'll move to your implementation as all sources can be treated as case 2 sources.